integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/CarbonScanRDD.scala

```diff
@@ -257,7 +257,8 @@ class CarbonScanRDD(
       cons.head.newInstance(queryModel).asInstanceOf[RecordReader[Void, Object]]
     } catch {
       case e: Exception =>
-        null
+        LOGGER.error(e)
+        throw e
```

Comment:
I think you can Log it and return null, it will create normal CarbonRecordRead in CarbonScanRDD